### ISSUE TITLE
Question: redux-thunk vs redux-saga.

### ISSUE THREAD

**Issue by gondar00** [verekia/js-stack-from-scratch#138]
### Type of issue: (feature suggestion, bug, translation?)
Question
### Chapter:
5
### If it's a bug:
No.
Please try using the code provided in this repository instead of your own to see if that solves the issue. If it does, compare the content of relevant files to see if anything is missing in your version. Every chapter is automatically tested, so issues are likely coming from missing an instruction in the tutorial, a typo, or something like this. Feel free to open an issue if there is a problem with instructions though.

Hi. Thank you for this amazing tutorial. 

Why have you chosen redux-thunk over redux-saga?

**Comment by verekia** [comment]
Good question :)

Generators might be a little too intense for people who are not well-versed in JS, and redux-thunk is kind of the "official" way to deal with async actions. The Redux chapter is already the most difficult one. I might replace it by redux-saga in the future if there is a strong demand though.

**Comment by gfortaine** [comment]
Did you consider [Redux Pack](https://github.com/lelandrichardson/redux-pack) ?

**Comment by Grsmto** [comment]
The fact that there is 2 different libraries proposed in 2 messages, answers by itself :)
Redux-thunk is the logical way to go for a beginner as it's in the Redux doc and most of the tutorials are using it.

**Comment by icflorescu** [comment]
...and here's a third candidate: [redux-promise](https://github.com/acdlite/redux-promise) :)

**Comment by roryokane** [comment]
And another option: [redux-loop].

Links to all options mentioned so far: 

* [redux-thunk]
* [redux-saga]
* [redux-pack]
* [redux-promise]
* [redux-loop]

[redux-thunk]: https://github.com/gaearon/redux-thunk
[redux-saga]: https://github.com/redux-saga/redux-saga
[redux-pack]: https://github.com/lelandrichardson/redux-pack
[redux-promise]: https://github.com/acdlite/redux-promise
[redux-loop]: https://github.com/redux-loop/redux-loop

**Comment by rbrtsmith** [comment]
I agree that Redux Thunk is the best choice for basic async workflow.  I don't see enough benefit to be using Redux Saga unless the async behaviour becomes quite complex.

**Comment by gfortaine** [comment]
Hello,

To enrich the current thinking, I would greatly appreciate to invite you to a further reading of the following post :

**Comment by gfortaine** [comment]
[Don't Fear The Fractal: Infinite State Composition With Freactal](http://formidable.com/blog/2017/infinite-state-composition-with-freactal/)

**Comment by jasonnathan** [comment]
I thought [redux-observable](https://redux-observable.js.org/) is a worthy alternative
